### ISSUE TITLE
Noise with Web App and frontdoor resources

### ISSUE THREAD

**Issue by MichelSkr** [Azure/arm-template-whatif#250]
**Describe the bug**
We have "Web APP" and "Frontdoor" Resources that are successfully created using bicep.
Whenever We run the "az deployment" command with "what-if" if shows parameters to be added again in Web APP resources and properties IDs to be deleted from frontdoor as shown in picture below

![](https://user-images.githubusercontent.com/87542713/155352155-d47f7f63-dd47-4b92-a016-4cffeb670d35.png)

**To Reproduce**
Every-time We run az deployment what-if We see that bicep wants to update the resources

**Expected behavior**
No changes to the resources since We didn't change the bicep templates 

**Screenshots**

**Client** [e.g. PowerShell, CLI, API)

**Additional context**

**Comment by stephaniezyen** [comment]
Part of this is a duplicate of #90 
The other part is a provider bug

**Comment by torreymicrosoft** [comment]
Thanks for your patience, this was recently picked up as one of the older open issues for the ARM Deployments team to revisit. Apologies for the long silence since 2022.

What-If for Azure Deployment Stacks went generally available today, and the headline feature is noise reduction, enabled in all regions. A stack records a what-if baseline when it is deployed, and any property that is unchanged between that baseline and the current evaluation is removed from the result as a false positive.

Your screenshot has three separate things in it and all three are that class:

- On `Microsoft.Web/sites`, the `+ properties.siteConfig.ftpsState`, `+ properties.siteConfig.localMySqlEnabled` and `+ properties.siteConfig.netFrameworkVersion` lines are App Service applying its own defaults and returning them on GET. I confirmed with Bicep 0.46.1 that all three are valid writable properties, so this is not an authoring mistake on your side.
- On `Microsoft.Network/frontDoors`, `- properties.cName` is a response only field: setting it is rejected outright by the current type definitions (BCP037 on `FrontDoorProperties`). The `- id` lines inside `backendPools`, `healthProbeSettings` and `loadBalancingSettings` are service generated ids in the same way.
- The app settings block reported as an addition is the same pattern on `Microsoft.Web/sites/config`.

None of it is a change your deployment would actually make, and all of it is identical run over run, which is the input the stack baseline is built to absorb.

You were running `az deployment`, so the stacks equivalent at resource group scope:

```
az stack-whatif group create \
  --name whatif-preview \
  --resource-group <resource-group> \
  --stack-id <deployment-stack-resource-id> \
  --template-file main.bicep \
  --action-on-unmanage detachAll \
  --deny-settings-mode none \
  --retention-interval PT3H
```

One honest caveat before you try it. The baseline is recorded when the stack is deployed or updated, so deploy or update the stack once to establish it, then run what-if against the stack. The docs are also clear that noise reduction removes many of these differences but not every one, so it is still worth reading the result rather than assuming everything left is real.

Announcement: https://techcommunity.microsoft.com/blog/azuregovernanceandmanagementblog/now-generally-available-what-if-for-azure-deployment-stacks/4547614
Docs: https://learn.microsoft.com/azure/azure-resource-manager/bicep/deployment-stacks-what-if

The App Service `siteConfig` half of this was split out on the thread back in 2022 and is tracked separately as #90. I am closing this issue out. If any of these properties survive a Stacks What-if run once the baseline is in place, reopen here or file in `Azure/deployment-stacks` with the what-if result resource ID.
